vortex-array/src/array/constant/compute.rs

```diff
@@ -59,11 +70,59 @@ impl SearchSortedFn for ConstantArray {
     }
 }
 
+impl AndFn for ConstantArray {
+    fn and(&self, array: &Array) -> VortexResult<Array> {
+        constant_array_bool_impl(self, array, |(l, r)| l & r, and)
+    }
+}
+
+impl OrFn for ConstantArray {
+    fn or(&self, array: &Array) -> VortexResult<Array> {
+        constant_array_bool_impl(self, array, |(l, r)| l | r, or)
+    }
+}
+
+fn constant_array_bool_impl(
+    constant_array: &ConstantArray,
+    other: &Array,
+    bool_op: impl Fn((bool, bool)) -> bool,
+    fallback_fn: impl Fn(&Array, &Array) -> VortexResult<Array>,
+) -> VortexResult<Array> {
+    if constant_array.len() != other.len() {
+        vortex_bail!("Boolean operations aren't supported on arrays of different lengths")
+    }
+
+    if !constant_array.dtype().is_boolean() || !other.dtype().is_boolean() {
+        vortex_bail!("Boolean operations are only supported on boolean arrays")
+    }
+
+    // If the right side is constant
+    if let Some(true) = other.statistics().compute_is_constant() {
```

Comment:
For now we should get the stat only if it's present. Calculation is a bit naive and might be as expensive as running the function.